cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -55,6 +56,32 @@ visualization_msgs::Marker CreateTrajectoryMarker(const int trajectory_id,
   return marker;
 }
 
+int GetLandmarkIndex(const std::string& landmark_id) {
+  static auto* landmark_id_to_index = new std::unordered_map<std::string, int>;
+  auto it = landmark_id_to_index->find(landmark_id);
+  if (it == landmark_id_to_index->end()) {
+    const int new_index = landmark_id_to_index->size();
+    landmark_id_to_index->at(landmark_id) = new_index;
+    return new_index;
+  }
+  return it->second;
+}
+
+visualization_msgs::Marker CreateLandmarkMarker(const std::string& landmark_id,
+                                                const Rigid3d& landmark_pose,
+                                                const std::string& frame_id) {
+  visualization_msgs::Marker marker;
+  marker.ns = "Landmarks";
+  int landmark_index = GetLandmarkIndex(landmark_id);
```

Comment:
Why not just hash the landmark_id?

Reply:
Hashes can have collisions.

Reply:
I think that using a string for an ID doesn't mesh well with the rest of the cartographer codebase - all other IDs (submap, node) are integer-based.